src/main/java/seedu/address/model/deliverymodel/DeliveryBook.java

```diff
@@ -108,6 +108,8 @@ public void removeDelivery(Delivery key) {
      * Copies the current {@code InventoryBook} in the state list.
      */
     public void commit() {
+        assert deliveryBookStatePointer < deliveryBookStateList.size();
+
         if (!deliveryBookStateList.isEmpty()) {
             deliveryBookStateList = deliveryBookStateList.subList(0, deliveryBookStatePointer + 1);
         }
```

Comment:
I feel that similarly to the command Hisotry there should be a limit of how many times u can redo incase of memory overflow

Reply:
Like limit the number of state stateList holds

Reply:
Good point. Will add it